pareto/examples/integrated_desalination_jupyter_notebooks/integrated_optimization_md_jupyter_notebook.ipynb

```diff
@@ -0,0 +1,210 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "d5e846f8",
+   "metadata": {},
+   "source": [
+    "## Integrating rigorous desalination models into network optimization\n",
+    "It is often desired to design a desalination plant at a certain desalination site in the network. However, it is not clear what the optimal design and operation of the plant should be. In this notebook we show how to co-optimize the network operation and desalination design by using the integrated optimization model available in PARETO. \n",
+    "\n",
+    "First we import the necessary utilities to read the network data, the function to build the integrated model with MD (Membrane Distillation) as the desalination model choice (`integrated_model_build`) and the Pyomo package."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7fc4dc81",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#####################################################################################################\n",
+    "# PARETO was produced under the DOE Produced Water Application for Beneficial Reuse Environmental\n",
+    "# Impact and Treatment Optimization (PARETO), and is copyright (c) 2021-2024 by the software owners:\n",
+    "# The Regents of the University of California, through Lawrence Berkeley National Laboratory, et al.\n",
+    "# All rights reserved.\n",
+    "#\n",
+    "# NOTICE. This Software was developed under funding from the U.S. Department of Energy and the U.S.\n",
+    "# Government consequently retains certain rights. As such, the U.S. Government has been granted for\n",
+    "# itself and others acting on its behalf a paid-up, nonexclusive, irrevocable, worldwide license in\n",
+    "# the Software to reproduce, distribute copies to the public, prepare derivative works, and perform\n",
+    "# publicly and display publicly, and to permit others to do so.\n",
+    "#####################################################################################################\n",
+    "\n",
+    "from importlib import resources\n",
+    "from pareto.utilities.get_data import get_data\n",
+    "from pareto.models_extra.CM_module.set_param_list import set_list, parameter_list\n",
```

Comment:
`CM_module.set_param_list` is being removed in #291 in favor of a new method for loading the set and parameter lists. So after that PR merges, we will need to make these Jupyter notebooks compatible with those changes.

Reply:
Addressed in latest commits - resolving.